run.py

```diff
@@ -197,13 +201,160 @@ def get_files(args, dirname, min_size=1, min_date=''):
+
+        for fp in file_paths:
+            actual_subset_size = 0
+
+            with gzip.open(fp, 'rt') as f:
+                # Create an output file handle
+                output_file = fp.replace(".fq.gz", ".subset.fq")
+                with open(output_file, "w") as out_handle:
+                    for index, (title, seq, qual) in enumerate(FastqGeneralIterator(f)):
+                        if index >= subset_size:
+                            break
+                        actual_subset_size += 1
+                        out_handle.write("@%s\n%s\n+\n%s\n" % (title, seq, qual))
+
+                output_files.append(output_file)
+
+        return output_files, total_reads, actual_subset_size
+
+
+    for sample in get_samples(args):
+        # Check for name of output file
+        sample_output_file = sample + ".ribocounts.txt"
+        if sample_output_file in existing_outputs: continue
+
+        total_reads_dict = {}
+        subset_reads_dict = {}
+        rrna_reads_dict = {}
+        for potential_input in available_inputs:
+            if not potential_input.startswith(sample): continue
+            if ".settings" in potential_input: continue
+            if "discarded" in potential_input: continue
+
+            # Number of output and input files must match 
+            tmp_fq_output = potential_input.replace(".gz", ".subset.nonrrna.fq")
+            tmp_fq_outputs = [tmp_fq_output]
+            inputs = [potential_input]
+
+            if ".pair1." in potential_input:
+                tmp_fq_outputs.append(tmp_fq_output.replace(".pair1.", ".pair2."))
+                inputs.append(potential_input.replace(".pair1.", ".pair2."))
+            elif ".pair2." in potential_input:
+                # Ribodetector handles pair1 and pair2 together.
+                continue
+
+            with tempdir("ribocounts", sample + " inputs") as workdir:
+                for input_fname in inputs:
+                    subprocess.check_call([
+                        "aws", "s3", "cp", "%s/%s/cleaned/%s" % (
+                            S3_BUCKET, args.bioproject, input_fname), input_fname])
+
+                    # Ribodetector gets angry if the .fq extension isn't in the filename
+                    os.rename(input_fname, input_fname.replace(".gz", ".fq.gz"))
+
+                # Add .fq extensions to input files
+                inputs = [i.replace(".gz", ".fq.gz") for i in inputs]
+
+                # Get subset of inputs
+                subsets, total_reads, subset_reads  = first_subset_fastq(inputs, subset_size)
+                subset_reads_dict[inputs[0]] = subset_reads
+                total_reads_dict[inputs[0]] = total_reads
+                
+                # Compute average read lengths. For paired-end reads, average length is
+                # computed only from pair1 reads.
+                print("Calculating average read length...")
+                def calculate_average_read_length(file_path):
```

Comment:
why not run this on the subsets?

Reply:
I am running it on the subsets. See line 296.

`avg_length = calculate_average_read_length(subsets[0])`